docs/essentials/themes.md

```diff
@@ -0,0 +1,88 @@
+---
+title: 'Themes'
+---
+
+Storybook's [Themes](https://github.com/storybookjs/storybook/tree/next/code/addons/themes) addon allows you to switch between multiple themes for your components inside of the preview in [Storybook](https://storybook.js.org).
+
+![Switching between themes in Storybook](./addon-themes-example.gif)
+
+## Theme decorators
+
+To make your themes accessible to your stories, `@storybook/addon-themes` exposes three [decorators](https://storybook.js.org/docs/react/writing-stories/decorators) for different methods of theming
+
+### JSX providers
+
+For libraries that expose themes to components through providers, such as [Material UI](https://storybook.js.org/recipes/@mui/material/), [Styled-components](https://storybook.js.org/recipes/styled-components/), and [Emotion](https://storybook.js.org/recipes/@emotion/styled/), use the `withThemeFromJSXProvider`.
+
+```ts
+// .storybook/preview.ts
+import type { Renderer } from '@storybook/your-framework'
+import { withThemeFromJSXProvider } from '@storybook/addon-themes';
+
+import { createGlobalStyle, ThemeProvider } from 'styled-components';
+import { lightTheme, darkTheme } from '../src/themes';
+
+/* snipped for brevity */
+
+export const decorators = [
+  withThemeFromJSXProvider<Renderer>({
+  themes: {
+    light: lightTheme,
+    dark: darkTheme,
+  }
+  defaultTheme: 'light',
+  Provider: ThemeProvider,
+  GlobalStyles,
+})];
+```
```

Comment:
@Integrayshaun, can you convert the snippets documented here into two files, specifically one for JavaScript and one for TypeScript, and place them inside the `common` directory?
Also, if you're ok with it when converting the examples, can you use the new structure that's in use throughout the documentation so that we don't confuse our readers and drop the `snipped for brevity` as someone might take it the wrong way?

Reply:
Appreciate it 🙇